ext/kv/01_db.ts

```diff
@@ -283,20 +275,16 @@ class Kv {
         } catch (error) {
           console.error("Exception in queue handler", error);
         } finally {
-          if (this.#closed) {
-            core.close(handleId);
-          } else {
-            const promise: Promise<void> = core.opAsync(
-              "op_kv_finish_dequeued_message",
-              handleId,
-              success,
-            );
-            finishMessageOps.set(handleId, promise);
-            try {
-              await promise;
-            } finally {
-              finishMessageOps.delete(handleId);
-            }
+          const promise: Promise<void> = core.opAsync(
```

Comment:
Is the removal of `this.#closed` correct?

If `close()` is called and then the `finally` block is reached, `op_kv_finish_dequeued_message` would throw because it received an invalid rid.

Reply:
`close()` closes the db rid. Each in-flight message has its own rid.

Previously we needed `this.#closed` because rust implementation of `op_kv_finish_dequeued_message` threw when called on closed database. Now that path is handled gracefully: https://github.com/denoland/deno/blob/main/ext/kv/sqlite.rs#L401, so we don't need to handle it in javascript.

Reply:
But `op_kv_dequeue_next_message` uses the db rid. It is an asynchronous operation - is it strictly fenced against a concurrent `close()`?

Reply:
You're right! I'm going to update `op_kv_dequeue_next_message` to fail gracefully if the db rid is dropped (instead of relying on `this.#closed`).